tools/unicode_table_generator.d

```diff
@@ -146,20 +166,56 @@ auto fullCaseEntry(dstring value, ubyte num, ubyte batch_size)
 }
 
 enum {
-    UnicodeDatabaseDirectory = "ucd-15/",
-    caseFoldingSrc = UnicodeDatabaseDirectory ~ "CaseFolding.txt",
-    blocksSrc = UnicodeDatabaseDirectory ~ "Blocks.txt",
-    propListSrc = UnicodeDatabaseDirectory ~ "PropList.txt",
-    graphemeSrc = UnicodeDatabaseDirectory ~ "auxiliary/GraphemeBreakProperty.txt",
-    emojiDataSrc = UnicodeDatabaseDirectory ~ "emoji/emoji-data.txt",
-    propertyValueAliases = UnicodeDatabaseDirectory ~ "PropertyValueAliases.txt",
-    corePropSrc = UnicodeDatabaseDirectory ~ "DerivedCoreProperties.txt",
-    normalizationPropSrc = UnicodeDatabaseDirectory ~ "DerivedNormalizationProps.txt",
-    scriptsSrc = UnicodeDatabaseDirectory ~ "Scripts.txt",
-    hangulSyllableSrc = UnicodeDatabaseDirectory ~ "HangulSyllableType.txt",
-    unicodeDataSrc = UnicodeDatabaseDirectory ~ "UnicodeData.txt",
-    compositionExclusionsSrc = UnicodeDatabaseDirectory ~ "CompositionExclusions.txt",
-    specialCasingSrc = UnicodeDatabaseDirectory ~ "SpecialCasing.txt"
+    caseFoldingSrc = "CaseFolding.txt",
+    blocksSrc = "Blocks.txt",
+    propListSrc = "PropList.txt",
+    graphemeSrc = "auxiliary/GraphemeBreakProperty.txt",
+    emojiDataSrc = "emoji/emoji-data.txt",
+    propertyValueAliases = "PropertyValueAliases.txt",
+    corePropSrc = "DerivedCoreProperties.txt",
+    normalizationPropSrc = "DerivedNormalizationProps.txt",
+    scriptsSrc = "Scripts.txt",
+    hangulSyllableSrc = "HangulSyllableType.txt",
+    unicodeDataSrc = "UnicodeData.txt",
+    compositionExclusionsSrc = "CompositionExclusions.txt",
+    specialCasingSrc = "SpecialCasing.txt"
+}
+
+void ensureFilesAreDownloaded()
+{
+    import std.file : write, mkdirRecurse;
+    import std.net.curl : download;
+    import std.path : dirName, buildPath;
+
+    string[] files = [
+        caseFoldingSrc,
+        blocksSrc,
+        propListSrc,
+        graphemeSrc,
+        emojiDataSrc,
+        propertyValueAliases,
+        corePropSrc,
+        normalizationPropSrc,
+        scriptsSrc,
+        hangulSyllableSrc,
+        unicodeDataSrc,
+        compositionExclusionsSrc,
+        specialCasingSrc
+    ];
+
+    mkdirRecurse(unicodeDir);
+    foreach (string file; files)
```

Comment:
...and then just `foreach` over the `EnumMembers` of it.

Reply:
I considered that, but I prefer to avoid advanced D features in scripts, I think it's easier to read that way.